src/economy/roadbase.cc

```diff
@@ -92,9 +158,23 @@ void RoadBase::set_path(EditorGameBase& egbase, const Path& path) {
 	idle_index_ = path.get_nsteps() / 2;
 }
 
-static inline void
-set_roadtype(EditorGameBase& egbase, const FCoords curf, uint8_t dir, RoadType type) {
+void RoadBase::set_roadtype(EditorGameBase& egbase,
+                            const FCoords curf,
+                            uint8_t dir,
+                            RoadType type) const {
 	if (dir == WALK_SW || dir == WALK_SE || dir == WALK_E) {
+		if (type != RoadType::kNone && is_bridge(egbase, curf, dir)) {
```

Comment:
Here, if I understand, you are modifying type (that is ROAD type to bridge type (if the section is/can be bridge))

Reply:
The local variable `type` in this function is used only as an argument to `EditorGameBase::set_road`, which does nothing more than to modify map markings for visualisation. The road´s member variable `type_` (which is `kNormal` or `kBusy`) is not read out or modified in this function.

Reply:
So RoadType on input can be None, kNormal or kBusy, and we are changing it inside the method to kBridge* if needed...

Maybe we should need two enums, enum RoadType with 3 options and new one like VisualRoadType with 5 options (or 4)

Reply:
Implemented it like this.
I auctally removed the "logical" RoadType now as it´s not needed anymore; RoadSegment is now used only for rendering stuff.
As a side effect, it would now be easily possible to implement special markers for road-/waterway-building-mode and building entrances if we want to…

Reply:
looks good. I have no more comments to the code, but the change is really big...